protocol/auditor.go

```diff
@@ -0,0 +1,176 @@
+// This module implements a generic CONIKS auditor, i.e. the
+// functionality that clients and auditors need to verify
+// a server's STR history.
+
+package protocol
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/coniks-sys/coniks-go/crypto"
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+)
+
+// Auditor provides a generic interface allowing different
+// auditor types to implement specific auditing functionality.
+type Auditor interface {
```

Comment:
I agree that `ConsistencyChecks` and `AuditLog` should "extend" `Auditor` rather than `AudState`, but to me this problem is more about how we can express what we mean. 

Ideally, what I would have liked to do here is declare a generic auditor type/interface, as one would in Java for example, and have the `AuditLog` and `ConsistencyChecks` extend that class and implement something like an abstract `AuditDirectory` method. I think what we have now comes closest to this. Maybe the solution would be to rename `AudState` to `Auditor` and rename `Auditor` to something like `AbstractAuditor`?